File: packages/create-react-native-library/templates/java-library/android/src/main/java/com/{%- project.package %}/{%= project.name %}Module.java
package com.<%- project.package %>;

import androidx.annotation.NonNull;

import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.module.annotations.ReactModule;

@ReactModule(name = <%- project.name %>Module.NAME)
public class <%- project.name %>Module extends ReactContextBaseJavaModule {
    public static final String NAME = "<%- project.name %>";

    public <%- project.name %>Module(ReactApplicationContext reactContext) {
        super(reactContext);
    }

    @Override
    @NonNull
    public String getName() {
        return NAME;
    }

<% if (project.cpp) { -%>
    static {
        try {
            // Used to load the 'native-lib' library on application startup.
            System.loadLibrary("cpp");
        } catch (Exception ignored) {
        }
    }
<% } -%>

    // Example method
    // See https://reactnative.dev/docs/native-modules-android
    @ReactMethod
    public void multiply(int a, int b, Promise promise) {
<% if (project.cpp) { -%>
        promise.resolve(nativeMultiply(a, b));
<% } else { -%>
        promise.resolve(a * b);
<% } -%>
    }

    public static native int nativeMultiply(int a, int b);
}

File: templates/native-library/android/src/main/java/com/{%= project.package %}/{%= project.name %}Module.kt
package com.<%= project.package %>

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import com.facebook.react.bridge.Promise

class <%= project.name %>Module(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {

    override fun getName(): String {
        return "<%= project.name %>"
    }

    // Example method
    // See https://facebook.github.io/react-native/docs/native-modules-android
    @ReactMethod
    fun multiply(a: Int, b: Int, promise: Promise) {
    <% if (project.cpp) { %>
      promise.resolve(nativeMultiply(a, b));
    <% } else { %>
      promise.resolve(a * b)
    <% } %>
    }

    <% if (project.cpp) { %>
    external fun nativeMultiply(a: Int, b: Int): Int;

    companion object
    {

        // Used to load the 'native-lib' library on application startup.
        init
        {
            System.loadLibrary("cpp")
        }
    }
    <% } %>
}
